assets/js/googlesitekit/data/create-error-store.js

```diff
@@ -92,12 +101,35 @@ export function createErrorStore() {
 
 			case CLEAR_ERROR: {
 				const { baseName, args } = payload;
-				const key = generateErrorKey( baseName, args );
+				const newState = { ...state };
+				if ( baseName ) {
+					const key = generateErrorKey( baseName, args );
+					newState.errors = { ...( state.errors || {} ) };
+					delete newState.errors[ key ];
+				} else {
+					// @TODO: remove it once all instances of the legacy behavior have been removed.
+					newState.error = undefined;
+				}
 
-				const errors = { ...( state.errors || {} ) };
-				delete errors[ key ];
+				return newState;
+			}
 
-				return { ...state, errors };
+			case CLEAR_ERRORS: {
+				const { baseName } = payload;
+				const newState = { ...state };
+				if ( baseName ) {
+					newState.errors = { ...( state.errors || {} ) };
+					for ( const key in Object.keys( newState.errors ) ) {
+						if ( key === baseName || key.startsWith( baseName ) ) {
+							delete newState.errors[ key ];
+						}
+					}
+				} else {
+					// @TODO: remove it once all instances of the legacy behavior have been removed.
```

Comment:
This comment only applies to the `newState.error` assignment below, not to both assignments (the `errors` state property is the "correct" one to use while `error` is legacy).